### PR TITLE
Update self-service device management interface

### PR DESCRIPTION
**Pull request by vahancouver**
[SDKS-3572](https://pingidentity.atlassian.net/browse/SDKS-3572)
[iOS] Align Self-Service Device Management Interface with JS

### CODE REVIEW

**Review by rodrigoareis — APPROVED** [review]
Changes looks good to me

File: FRAuth/FRAuth/SelfService/DeviceClient.swift [hunk 1]
```diff
@@ -253,3 +212,49 @@ public class DeviceClient: DeviceRepository {
         return try JSONDecoder().decode(Session.self, from: data)
     }
 }
+
+
+/// Implementation of the `ImmutableDevice` protocol for managing devices that only support retrieval and deletion.
+@available(iOS 13.0.0, *)
+public struct ImmutableDeviceImplementation<R>: ImmutableDevice where R: Device {
+  var endpoint: String
```
Comment (george-bafaloukas-forgerock) [inline FRAuth/FRAuth/SelfService/DeviceClient.swift:220]: Ti my previous comment, I think that should be enough, as this is a Public struct and a devloper could add their values and set it. On that note, I suggest we add some code docs here on what values are expected for these properties

File: SampleApps/FRExample/FRExample/ViewControllers/ClientDevicesTableViewController.swift [hunk 2]
```diff
@@ -79,7 +79,18 @@ class ClientDevicesTableViewController: UITableViewController, AlertShowing {
             alert.addAction(UIAlertAction(title: "Yes", style: .destructive, handler: { (alert: UIAlertAction!) in
                 Task {
                     do {
-                        try await self.deviceRepo.delete(device: device)
+                        //try await self.deviceRepo.delete(device: device)
```
Comment (george-bafaloukas-forgerock) [inline SampleApps/FRExample/FRExample/ViewControllers/ClientDevicesTableViewController.swift:82]: Was that intentionally left commented out?

**Review by george-bafaloukas-forgerock — COMMENTED** [review]
LGTM, 2 minor comments. Vahan let me know if you want to discuss my comments suggestion further

**Review by jeyanthanperiyasamy — APPROVED** [review]

**Review by witrisna — APPROVED** [review]

**Review by spetrov — APPROVED** [review]
LGTM!

**Review by george-bafaloukas-forgerock — APPROVED** [review]
LGTM

**Pull request merged**